### PR TITLE
(2) feat: compute percentile statistic

### PR DESCRIPTION
**Pull request by jfdoming**
## Notion ticket link

[Add percentile scoring](https://www.notion.so/uwblueprintexecs/Add-percentile-scoring-311d6043a67d4e79ae489a295fd6f716)

## Implementation description
* Follow-up to #380 

## Steps to test
1. View an assessment drilldown and verify you see the correct percentile.

## What should reviewers focus on?
* Clean? Correct?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/graphql/resolvers/testSessionResolvers.ts [hunk 1]
```diff
@@ -68,7 +69,7 @@ const testSessionResolvers = {
       if (!results) return [];
 
       const { students } = await classService.getClassById(parentClass);
-      const resultsByStudentId = results.reduce<
+      const resultsByStudentId = computePercentiles(results, "score").reduce<
```
Comment (joyce-shi) [inline backend/graphql/resolvers/testSessionResolvers.ts:72]: do we re-compute percentile every time we query for student results?

**Review by joyce-shi — COMMENTED** [review]
just had one q

File: backend/graphql/resolvers/testSessionResolvers.ts (hunk 1 above)
Comment (jfdoming) [reply backend/graphql/resolvers/testSessionResolvers.ts:72]: Yes, I had remembered that we were deferring optimization until later. It's just a sort so I'm not too too worried about performance, but ultimately we should probably move it to the completion mutation or something. Do you think we should do that now?

Comment (joyce-shi) [reply backend/graphql/resolvers/testSessionResolvers.ts:72]: all good. approving!

**Review by joyce-shi — APPROVED** [review]

**Pull request merged**